text/0000-lang-team-mcp.md

```diff
@@ -0,0 +1,216 @@
+
+This should help with the problem that longer RFCs threads can be quite hard to follow. This is compounded by the fact that the RFC evolves over the course of the discussion, sometimes changing dramatically. Getting "up to speed" on the state of an RFC thread involves reading a mix of stale comments on older drafts that have long since been addressed, thoughts on motivation, details of the design, and other things, and trying to sort them into an order.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Hello! If you would like propose a change to the Rust language, you've come to the right place. The process can be summarized as follows (but check out the [flowchart] for a graphical version):
+
+* Open a **major change proposal** (MCP) issue on the lang-team repository
+    * The issue should summarize the motivation and sketch out some details of your idea.
+    * You don't have to have everything figured out at this stage, and it's more important to emphasize the problems you are solving and why they're important than the nitty gritty details of how to solve them.
+    * When you create a proposal, we will also automatically make a Zulip stream for discussion. The issue thread is reserved for summaries, announcements, or procedural updates.
+* If the MCP is simple enough, the team may direct you to go straight to implementation.
+    * In that case, you can create a PR against the rust-lang/rust repository.
+* Otherwise, the proposal will require chartering a project group and drafting an RFC. This step requires a **lang-team liaison** who is willing and able to help the idea move forward.
+    * The first step will be chartering a project group, which the liaison will help you with.
+    * The charter is very similar to the MCP itself, and describes the motivation and other considerations that may have come up.
+* Of course, many proposals will not be accepted. This can occur for a number of reasons:
+    * There are no lang team members who have available bandwidth to serve as a liaison right now.
+    * The problem being solved doesn't seem important enough or is not a match for Rust's current priorities.
+    * The proposal has flaws in its design that don't seem like they can be solved.
+    
+## Flowchart
+
+You can also view the [flowchart in graphical form][flowchart]. This
+is derived from a [mermaid document][flowchart-source] and would be
+posted on the lang-team website.
+
+[flowchart]: https://is.gd/nMrsVE
+[flowchart-source]: https://is.gd/LdRVWk
+
+## Reasons to accept or decline a proposal
```

Comment:
I think one of the most intimidating things about going through the MCP/RFC process is not having clarity on where a proposal stands between the "opening" and "accept/reject" phase. In fact, as this text mentions, many RFCs are stuck in limbo between these two phases. This RFC as currently written helps avoid the "indefinite limbo" situation, but not necessarily the lack of clarity leading up to an outcome.

Decisions sometimes take time, and we can't expect to give everyone 100% insight into how a proposal is going to turn out. However, we can provide more clarity around the process itself: when an MCP will be discussed (e.g. the next lang team meeting), why it might not get discussed (e.g. the agenda is full and it gets kicked to the following meeting), and what the possible outcomes of that discussion are (acceptance, rejection, deferral to another meeting once members have had time to process). Ideally, we can also provide process-related updates as comments on the MCP, but that would increase the overhead of managing the lang team.

Reply:
I'm going to add a note that we will review proposals during our weekly triage meetings. I think the hope is that we will be much more aggressive in closing proposals that we don't plan to act on **now**. With RFCs, we often left proposals open for some time to let them "bake" and gather feedback. The intention here I think is that since the goal of an MCP is to seek a lang-team liaison most of all, we should be able to respond more promptly. We'll see how that goes.

Reply:
I think I should probably clarify this intent as well.